### PR TITLE
Dialogue: improve speakers autocomplete handling

### PR DESCRIPTION
**Pull request by retrofox**
This PR improves the way of handling the speaker's autocompleter.

Fixes #

#### Changes proposed in this Pull Request:

* Dialogue: improve speakers autocomplete handling

#### Jetpack product discussion

> When the suggestion text is the same as the label I’ve already entered, and there’s just one suggestion, I wouldn’t expect to see suggestions at all:
> 

p7DVsv-atS-p2#comment-34461

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Test the whole implementation gets a little bit tricky. The best way to do it is by paying attention to the CSS class defined in the participant element:

State | Canvas | Element Inspector
-----|------|-----
Create a new block ( **ready** - initial state) | image | image
Start to type to add anew participant (**adding**) | image | image
Add the new participant (**added**) | image | image
Start to edit (**editing**) | image | image
Edit the participant (**edited**) | image | image
When editing the label matches with a participant already added, it changes to **selecting** | image | image
Once **selected** | image | image

In short, confirm that the Autocompleter is only shown when adding a new speaker, or by selecting while typing a new one and it matches with a label already added.

#### Proposed changelog entry for your changes:

*

### CODE REVIEW

**Review by allilevine — APPROVED** [review]
Works as described in Firefox and Chrome. ✅

**Review by jeherve — APPROVED** [review]
This seems to test well for me. 🚢

**Comment by retrofox** [comment]
r222111-wpcom
